monai/data/csv_saver.py

```diff
@@ -26,44 +25,45 @@ class CSVSaver:
     Typically, the data can be classification predictions, call `save` for single data
     or call `save_batch` to save a batch of data together, and call `finalize` to write
     the cached data into CSV file. If no meta data provided, use index from 0 to save data.
+    Note that this saver can't support multi-processing because it reads / writes single
+    CSV file and can't guarantee the data order in multi-processing situation.
+
     """
 
     def __init__(self, output_dir: str = "./", filename: str = "predictions.csv", overwrite: bool = True) -> None:
         """
         Args:
             output_dir: output CSV file directory.
             filename: name of the saved CSV file name.
-            overwrite: whether to overwriting existing CSV file content. If we are not overwriting,
-                then we check if the results have been previously saved, and load them to the prediction_dict.
+            overwrite: whether to overwriting existing CSV file content, if True, will clear the file before saving.
+                otherwise, will apend new content to the CSV file.
 
         """
         self.output_dir = output_dir
         self._cache_dict: OrderedDict = OrderedDict()
         if not (isinstance(filename, str) and filename[-4:] == ".csv"):
```

Comment:
this filename extension constraint is restrictive, do we really need this check? perhaps a warning?

Reply:
sure, changed to a warning in the latest commit.
Thanks.